packages/utils/src/surescripts/send-customer-request.ts

```diff
@@ -0,0 +1,103 @@
+import { writeSurescriptsRunsFile, appendToSurescriptsRunsFile } from "./shared";
+
+/**
+ * Sends a request for all patients of the given customer, batching requests by facility. This is the preferred method
+ * to accurately send a backfill request to Surescripts for all patients with the minimum number of SFTP transactions.
+ *
+ * Usage:
+ * npm run surescripts -- customer-request --cx-id <cx-id> --csv-output <csv-output> --batch-size <batch-size>
+ *
+ * cx-id: The CX ID of the requester
+ * csv-output: The file to write CSV IDs to
+ * batch-size: The maximum number of patients per batch request - defaults to 100
+ *
+ * Example:
+ * npm run surescripts -- customer-request --cx-id "acme-uuid-1234-sadsjksl" --csv-output "acme-roster.csv" --batch-size 100
+ *
+ * Note: The headers of the CSV output file are "facility_id","transmission_id","patient_id", which are used in scripts
+ * like `convert-customer-responses` to convert the eventual responses from Surescripts into FHIR resources.
+ */
+const program = new Command();
+
+program
+  .name("customer-request")
+  .requiredOption("--cx-id <cx>", "The CX ID of the requester")
+  .requiredOption("--csv-output <csvOutput>", "The file to write CSV IDs to")
+  .option("--batch-size <batchSize>", "The maximum number of patients per batch request", "100")
+  .description(
+    "Send a request to Surescripts for all patients of the given customer, batched by facility"
+  )
+  .showHelpAfterError()
+  .version("1.0.0")
+  .action(
+    async ({
+      cxId,
+      batchSizeString,
+      csvOutput,
+    }: {
+      cxId: string;
+      batchSizeString?: string;
+      csvOutput: string;
+    }) => {
+      const batchSize = parseInt(batchSizeString ?? "100");
+      if (!Number.isFinite(batchSize)) {
+        throw new Error("Batch size must be a number");
+      }
+      const { log } = out(`Surescripts sendCustomerRequest - cxId ${cxId}`);
+      const dataMapper = new SurescriptsDataMapper();
+      const batchRequests = await dataMapper.getBatchRequestDataByFacility(cxId, batchSize);
+      const facilityIds = Object.keys(batchRequests);
+      log(`Sending requests across ${facilityIds.length} facilities`);
+
+      const csvOutputPath = writeSurescriptsRunsFile(
+        csvOutput,
+        `"facility_id","transmission_id","patient_id"\n`
+      );
+      let totalRequestedPatients = 0;
+      let totalBatches = 0;
+
+      for (const facilityId of facilityIds) {
+        const batchesForFacility = batchRequests[facilityId];
+        if (!Array.isArray(batchesForFacility)) {
+          log(`No requests for facility ${facilityId}`);
+          continue;
+        }
+
+        for (const batchRequest of batchesForFacility) {
+          // Create a new client for each batch request
+          // (i.e. do not re-use the client, as per the SFTP client instructions)
+          const client = new SurescriptsSftpClient({
+            logLevel: "info",
+          });
```

Comment:
why are we creating a new client for every batch request? Can we have one client that lives at the module level / top level?

Reply:
Each SFTP transaction involves a connect/disconnect, after which the underlying `ssh2-sftp-client`'s guidance is to create a new SFTP client (search [the docs](https://www.npmjs.com/package/ssh2-sftp-client#org3570f3c) for "Don't reuse SFTP client objects")

Reply:
Nice, makes sense. nit; If you need to say "as per the SFTP client instructions" it probably warrants a link to the doc.